src/include/btree_inline.h

```diff
@@ -2115,15 +2141,36 @@ __wt_btcur_skip_page(
         return (0);
 
     WT_REF_LOCK(session, ref, &previous_state);
-    if ((previous_state == WT_REF_DISK || previous_state == WT_REF_DELETED ||
-          (previous_state == WT_REF_MEM && !__wt_page_is_modified(ref->page))) &&
-      __wt_ref_addr_copy(session, ref, &addr) && addr.ta.newest_stop_txn != WT_TXN_MAX &&
-      addr.ta.newest_stop_ts != WT_TS_MAX &&
+
+    if (previous_state == WT_REF_MEM && !__wt_page_is_modified(ref->page))
```

Comment:
There was [this](https://github.com/mongodb/mongo/commit/3f041b19cc07d6225b8928b2318a5334caa74175#diff-193af7edb5ca67bb184cf97a300bbee131d82041bc3da1da8fc4a41ff7035a31R2358) more in 7.0 when the ref is deleted:
```
        walk_skip_stats->total_del_pages_skipped++;
```

Reply:
Yes, those code changes are due to the fast truncate data format changes. As those conditions are not possible in 6.0, we do not need it.